flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricNames.java

```diff
@@ -42,8 +42,11 @@ private MetricNames() {
 	public static final String IO_NUM_BUFFERS_OUT_RATE = IO_NUM_BUFFERS_OUT + SUFFIX_RATE;
 
 	public static final String IO_CURRENT_INPUT_WATERMARK = "currentInputWatermark";
+	@Deprecated
 	public static final String IO_CURRENT_INPUT_1_WATERMARK = "currentInput1Watermark";
+	@Deprecated
 	public static final String IO_CURRENT_INPUT_2_WATERMARK = "currentInput2Watermark";
+	public static final String IO_CURRENT_INPUT_WATERMARK_PATERN = "currentInput%dWatermark";
```

Comment:
Could also be done with prefix.

Reply:
As previously, even if we re ordered it as prefix, patterns carry more information and are more concrete/well defined. For example, if you have prefix, does it mean that there is no suffix or other infix? Pattern mitigates this ambiguity.